feat: additional table generate

shukurov-aleksander opened the pull request.

shukurov-aleksander committed:
feat: additional table generate

kolyago-vladislav commented on databases/business/migrations/2023-02-26_intermediate-changelog.xml:
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
зачем тебе этот файл?

kolyago-vladislav commented on databases/business/migrations/test-data/2/2023-02-26__2_fill-table-companies.xml:
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
используй уже существующие скрипт на генерации тест даты в таблицу компани

kolyago-vladislav commented on databases/business/migrations/test-data/2/2023-02-26__1_fill-table-company_statuses.xml:
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
вынеси этот файл из папки ко всем скриптам генераций

kolyago-vladislav commented on databases/business/migrations/test-data/2/2023-02-26__1_fill-table-company_statuses.xml:
```diff
@@ -0,0 +1,25 @@
+    <changeSet id="1" author="Aleksander.shukurov@gmail.com">
+        <comment>fill in the table "company_statuses" with test data</comment>
+        <sql>
+            with SERIES AS (
```
отформатируй нормально

kolyago-vladislav commented on databases/business/migrations/test-data/2/2023-02-26__1_fill-table-company_statuses.xml:
```diff
@@ -0,0 +1,25 @@
+                NOW() - FLOOR((RANDOM() * (1000 - 500) + 500)) * INTERVAL '1 day',
+                lpad('comment about company status', (FLOOR(RANDOM() * ((248 - 32) + 32)))::INTEGER, 'different comment and another'),
+                FLOOR(RANDOM() *((SELECT COUNT(*) FROM companies) - 1) + 1)
+            FROM series s ON CONFLICT DO NOTHING
```
ON CONFLICT DO NOTHING - это костыль в данном случае

shukurov-aleksander committed:
feat: fix update companies column "company_statuses_id"

kolyago-vladislav commented on databases/business/migrations/test-data/2/2023-02-26__2_fill-table-companies.xml:
```diff
@@ -5,9 +5,10 @@
     <changeSet id="1" author="Aleksander.shukurov@gmail.com">
         <comment>fill in the table "companies" column "company_statuses_id" with test data</comment>
         <sql>
```
отформатируй нормально

shukurov-aleksander committed:
fix: updated xml

Commit:
fix: fix test-data generation

Commit:
fix: fix findAll request

shukurov-aleksander committed:
fix: fill tables, added changes to companyRepository

shukurov-aleksander committed:
fix: replaced lambda with method()

shukurov-aleksander committed:
fix: refactoring

kolyago-vladislav commented on databases/business/migrations/1/2022-10-26__1_creating-enum-types.xml:
```diff
@@ -25,4 +25,15 @@
             </sql>
         </rollback>
     </changeSet>
+    <changeSet id="3" author="Aleksander.shukurov@gmail.com">
```
вынеси это в папку "2"

kolyago-vladislav commented on databases/business/migrations/1/2022-10-26__2_create-companies.xml:
```diff
@@ -16,6 +16,22 @@
         </createTable>
     </changeSet>
+
+    <changeSet id="2" author="Aleksander.shukurov@gmail.com">
```
аналогично в папку 2

kolyago-vladislav commented on databases/business/migrations/1/2023-02-26__2_create-company-statuses.xml:
```diff
@@ -0,0 +1,22 @@
+                <constraints primaryKey="true"/>
+            </column>
+            <column name="company_status" type="company_status_enum">
+                <constraints nullable="false" unique="true"/>
```
сделай создание юника отдельным ченжсетом

kolyago-vladislav commented on databases/business/migrations/1/cumulative-changelog.xml:
```diff
@@ -3,6 +3,7 @@
                    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
                    xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog https://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.8.xsd">
     <include file="2022-10-26__1_creating-enum-types.xml" relativeToChangelogFile="true"/>
+    <include file="2023-02-26__2_create-company-statuses.xml" relativeToChangelogFile="true"/>
```
тебя не смутило что индекс у миграции 2 уже существует?

kolyago-vladislav commented on src/main/java/com/ku/business/controller/CompanyController.java:
```diff
@@ -31,6 +32,8 @@ public List<CompanyListDto> findAll(
             @RequestParam(value = "user_id", required = false)  Long userId,
             @Parameter(description = "is government agency", example = "false")
             @RequestParam(value = "is_government_agency", required = false)  Boolean isGovernmentAgency,
+            @Parameter(description = "Company status", example = "REGISTERED")
```
добавь, пожалуйста, allowableValues как-то так, там стринга вроде будет делаешь так = "[REGISTERED, FREEZED]"

kolyago-vladislav replied:
а чем Parametr отличает от Схемы? надо что-то одно оставить

shukurov-aleksander replied:
done

kolyago-vladislav commented on src/main/java/com/ku/business/controller/CompanyController.java:
```diff
@@ -31,6 +32,8 @@ public List<CompanyListDto> findAll(
             @Parameter(description = "is government agency", example = "false")
             @RequestParam(value = "is_government_agency", required = false)  Boolean isGovernmentAgency,
+            @Parameter(description = "Company status", example = "REGISTERED")
+            @RequestParam(value = "company_status", required = false) CompanyStatus companyStatus,
```
не заметил сразу, companyStatus, userId исправь везде

shukurov-aleksander replied:
done

kolyago-vladislav commented on src/main/java/com/ku/business/repository/CompanyRepository.java:
```diff
@@ -1,34 +1,59 @@
-        SELECT *
+        SELECT 
+            c.id, 
+            c.company_name as company_name, 
```
as насколько я знаю можешь не писать

shukurov-aleksander replied:
done

kolyago-vladislav commented on src/main/java/com/ku/business/repository/CompanyRepository.java:
```diff
@@ -1,34 +1,59 @@
 
     public List<Company> findAll(CompanyFilter filter) {
-        return namedParameterJdbcTemplate.query(QUERY_FIND_ALL_WITH_FILTER, createMapOfFilteredFields(filter), (rs, rowNum) -> createCompanyFromResultSet(rs));
+        return namedParameterJdbcTemplate.query(
```
почему ты перестал использовать этот метод?
я просил сделать так:
это в лямде:
this::createCompanyFromResultSet
это в сигнатуре метода
public Company createCompanyFromResultSet(ResultSet rs, int rowNum) {

kolyago-vladislav replied:
метод можешь назвать buildCompanyListDto

kolyago-vladislav replied:
я думаю, в одну строку всё должно влезть FIND_ALL_QUERY, переименуй название запроса

shukurov-aleksander replied:
done

kolyago-vladislav commented on src/main/java/com/ku/business/repository/CompanyRepository.java:
```diff
@@ -38,21 +63,24 @@ public Company createCompanyFromResultSet(ResultSet rs) {
+                .addValue("isGovernmentAgencyNull", filter.getIsGovernmentAgency() == null)
+                .addValue("isGovernmentAgency", filter.getIsGovernmentAgency())
+                .addValue("isCompanyStatusNull", filter.getCompanyStatus() == null)
+                .addValue("companyStatus", filter.getCompanyStatus() != null ? filter.getCompanyStatus().toString() : "")
```
filter.getCompanyStatus так разве работать не будет?

shukurov-aleksander replied:
done

Commit:
fix:
migration tables creat and update;
lambda expression in findAll method of CompanyRepository class;
added available values;
refactoring

kolyago-vladislav commented on databases/business/migrations/1/2022-10-26__2_create-companies.xml:
```diff
@@ -16,6 +16,9 @@
             </column>
             <column name="user_id" type="bigint"/>
             <column name="is_government_agency" type="boolean"/>
+            <column name="company_status_id" type="bigint">
```
ты не можешь изменять существующие миграции, вынеси это в миграцию в папке 2

kolyago-vladislav replied:
чтобы проверить это, попробуй катнуть из мастера базу, а потом катнуть из твоей ветки, тебе будет писать, что-то типо хэш ченж сета изменён

shukurov-aleksander replied:
done

kolyago-vladislav commented on databases/business/migrations/test-data/2022-10-26__1_fill-table-companies.xml:
```diff
@@ -7,13 +7,14 @@
                    CASE
                        WHEN s.sery % 100 = 0 THEN TRUE ELSE FALSE
-                   END
+                       END,
```
END под CASE должен быть

shukurov-aleksander replied:
done

kolyago-vladislav commented on src/main/java/com/ku/business/repository/CompanyRepository.java:
```diff
@@ -10,49 +11,58 @@
-        SELECT *
+        SELECT c.id, c.company_name, c.tax_number, c.is_government_agency, c.user_id, cs.company_status
         FROM companies c
+        LEFT JOIN company_statuses cs on c.company_status_id = cs.id
```
лефт джоин часть фрома добавь один таб

shukurov-aleksander replied:
done

kolyago-vladislav commented on src/main/java/com/ku/business/repository/CompanyRepository.java:
```diff
@@ -10,49 +11,58 @@
 
     @SneakyThrows
-    public Company createCompanyFromResultSet(ResultSet rs) {
+    public Company buildCompanyListDto(ResultSet rs, int rowNum) {
```
у тебя метод называется buildCompanyListDto - а ты возвращаешь Company - entity
собирай сразу дто объект

shukurov-aleksander replied:
done

kolyago-vladislav commented on src/main/java/com/ku/business/repository/CompanyRepository.java:
```diff
@@ -10,49 +11,58 @@
+                .addValue("isGovernmentAgencyNull", filter.getIsGovernmentAgency() == null)
+                .addValue("isGovernmentAgency", filter.getIsGovernmentAgency())
+                .addValue("isCompanyStatusNull", filter.getCompanyStatus() == null)
+                .addValue("companyStatus", Objects.toString(filter.getCompanyStatus(),""))
```
сюда необязательно передавать ,"" можешь удалить

shukurov-aleksander replied:
done

Commit:
fix: refactoring

kolyago-vladislav commented on databases/business/migrations/2/2023-02-26__3_adding-unique-constraint-to-companies.xml:
```diff
@@ -0,0 +1,29 @@
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog https://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.8.xsd">
+    <changeSet id="1" author="Aleksander.shukurov@gmail.com">
+        <comment>add column "company_status_id" to "companies"</comment>
+        <sql>
```
почему ты через sql создаёшь колонку? 
addColumn - посмотри в xml есть такая команда

shukurov-aleksander replied:
done

Commit:
fix: changed column creation from sql to xml

Commit:
fix: refactoring

Commit:
fix: added nullable constraint to "company_status_id"
